### PR TITLE
Adjust MonitorEfficiencyCorUser for other instruments

### LINKED ISSUE #20924: MonitorEfficiencyCorUser: adjust to work with other instruments
`MonitorEfficiencyCorUser` should be adjusted to work with other instrument. Floolwing needs to be done:
1. Restriction to TOFTOF must be removed
2. Name of the monitor counts sample log should be read from the instrument parameter file.

please, assign me
milestone: 3.12

### PR DESCRIPTION
**Pull request by mganeva**
This PR removes restriction to TOFTOF instrument and modifies algorithm to read the name of the 'monitor counts' sample log from the instrument parameter file.

This makes possible to use `MonitorEfficiencyCorUser` for other instruments.

**To test:**
DNS and TOFTOF instruments store monitor counts not as a separate spectrum, but just as a single number stored as a sample log. For TOFTOF data the algorithm should work as before. To test with DNS data ([testdata.zip](https://github.com/mantidproject/mantid/files/1390159/testdata.zip)), the following script may be used:

```python
import numpy as np

# load raw data
ws_raw = LoadDNSLegacy("service0000409009vana.d_dat", Normalization='no')
# normalize to corrected monitor counts
ws1 = MonitorEfficiencyCorUser(ws_raw)

# load and normalize to raw monitor counts
ws2 = LoadDNSLegacy("service0000409009vana.d_dat", Normalization='monitor')

# calculate ratio, it should be sqrt(Ei/25.3) for non-zero counts
ratio = ws2/ws1

# compare to sqrt(Ei/25.3)
mcounts =  ws_raw.getRun().getLogData('mon_sum').value
Ei = ws_raw.getRun().getLogData('Ei').value
print "Correction coeff: {}, Ratio: {}".format(np.sqrt(Ei/25.3), ratio.readY(0)[5])
```

Fixes #20924. 

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [x] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [x] Do changes function as described? Add comments below that describe the tests performed?
- [x] Do the changes handle unexpected situations, e.g. bad input?
- [x] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/Algorithms/src/MonitorEfficiencyCorUser.cpp [hunk 1]
```diff
@@ -51,7 +46,19 @@ void MonitorEfficiencyCorUser::exec() {
...
+  // get name of the monitor counts sample log from the instrument parameter
+  // file
+  try {
+    mon_counts_log = getValFromInstrumentDef("monitor_counts_log");
+  } catch (Kernel::Exception::InstrumentDefinitionError) {
+    // the default value is monitor_counts
+    mon_counts_log = "monitor_counts";
+  }
+
+  Strings::convert(m_inputWS->run().getProperty(mon_counts_log)->value(),
```
Comment (soininen) [inline Framework/Algorithms/src/MonitorEfficiencyCorUser.cpp]: This looks awfully complicated. Have you tried using
```c++
m_monitorCounts = m_inputWS->run().getPropertyValueAsType<double>(mon_counts_log);
```
If the above works, you could make similar changes to the other similar cases within this algorithm.

File: docs/source/algorithms/MonitorEfficiencyCorUser-v1.rst [hunk 2]
```diff
@@ -11,22 +11,20 @@ Description
...
 
 The corrected value of the monitor counts is used to normalise the input workspace.
 
 
 Restrictions
 ###################################
 
-A formula named "formula\_eff" must be defined in the instrument
-parameters file. It is defined as "monitor counts * sqrt(Ei/25.3)"
+- A formula named "formula\_eff" must be defined in the instrument parameters file. For TOFTOF and DNS instruments it is defined as :math:`M*\sqrt{\frac{E_i}{25.3}}`. The incident energy :math:`E_i` and the monitor counts :math:`M` are read in the sample logs of the input workspace.
```
Comment (soininen) [inline docs/source/algorithms/MonitorEfficiencyCorUser-v1.rst]: May I suggest removing the '*' from the efficiency formula? Or try something like '\cdot' if you really want to show a multiplication operator there.

**Review by soininen — CHANGES_REQUESTED** [review]

**Comment by soininen** [comment]
@mganeva The test script provided with the PR doesn't run with the latest changes anymore. I think you should switch back to `int` type as the monitor counts log. With the changes in this diff:
[MonitorEfficiencyCorUser-int-monitor-log.diff.zip](https://github.com/mantidproject/mantid/files/1426671/MonitorEfficiencyCorUser-int-monitor-log.diff.zip) things seems to be working. Maybe you could have a look?

**Comment by mganeva** [comment]
@soininen Sorry for delay. If I will get back to int, `MonitorEfficiencyCorUser` will get in trouble with workspaces processed by `MergeRuns`. I decided to make monitor counts float also for DNS. 

Now the example script should work.

**Review by soininen — APPROVED** [review]

**Comment by soininen** [comment]
The test script seems to run now without problems.

**Pull request merged**